context.go

```diff
@@ -666,19 +666,15 @@ func (c *Context) ShouldBindWith(obj interface{}, b binding.Binding) error {
 // NOTE: This method reads the body before binding. So you should use
 // ShouldBindWith for better performance if you need to call only once.
 func (c *Context) ShouldBindBodyWith(obj interface{}, bb binding.BindingBody) (err error) {
-	var body []byte
-	if cb, ok := c.Get(BodyBytesKey); ok {
-		if cbb, ok := cb.([]byte); ok {
-			body = cbb
-		}
-	}
-	if body == nil {
-		body, err = ioutil.ReadAll(c.Request.Body)
-		if err != nil {
-			return err
-		}
-		c.Set(BodyBytesKey, body)
+	var buf bytes.Buffer
+	tee := io.TeeReader(c.Request.Body, &buf)
```

Comment:
This results in frequent buffer reads, which leads to lower efficiency. If ShouldBindBodyWith is used, subsequent code should only use ShouldBindBodyWith continuously, without sacrificing performance for convenience

Reply:
You're right if you call ShouldBindBodyWith frequently